operator/v1/types_dns.go

```diff
@@ -101,6 +100,31 @@ type DNSSpec struct {
 	// +optional
 	// +kubebuilder:default=Normal
 	LogLevel DNSLogLevel `json:"logLevel,omitempty"`
+
+	// cache describes the caching configuration that applies to all server blocks listed in the Corefile.
+	// This field allows a cluster admin to optionally configure:
+	// * successTTL which is a duration for which positive responses should be cached.
+	// * denialTTL which is a duration for which negative responses should be cached.
+	// If this is not configured, OpenShift will configure success and denial caching with a default value that is
+	// subject to change. At the time of writing, the default successTTL is 900 seconds and the default denialTTL is 30
+	// seconds or as noted in respective Corefile for your version of OpenShift.
+	// +optional
+	Cache *DNSCache `json:"cache,omitempty"`
```

Comment:
Why is this a pointer? We tend to avoid pointers in configuration APIs as they harm discoverability

Do you need to be able to distinguish between a zero value and a nil value?

Reply:
+1

Reply:
Is the net effect of removing the pointer that the DNSSpec will get an empty object? e.g.
```
spec:
  cache: {}
```
If so, that's fine with me. At this level an empty object is something we can pivot on in the operator. We do need a distinction in the child values though; `0` is an intentional setting, `nil` is unset. Does that need affect what needs to be changed here?

Reply:
Yeah, the effect is that users will see the cache object and therefore be able to understand that's something they may want to configure.

If you need to distinguish between 0 and nil, then they need to remain as pointers, you'll need to explain what a 0 value means though as part of the documentation and then equally explain what an omitted/nil value is

Reply:
Fixed!